.github/workflows/main.yml

```diff
@@ -152,8 +164,20 @@ jobs:
         python Scripts/mxupdate.py ../resources/Materials/TestSuite/stdlib/upgrade --yes
         python Scripts/mxvalidate.py ../resources/Materials/Examples/StandardSurface/standard_surface_marble_solid.mtlx --verbose
         python Scripts/mxdoc.py ../libraries/pbrlib/pbrlib_defs.mtlx
+        cd ..
+        mkdir build/generate
```

Comment:
Since we're not consuming these generated shaders, it might make more sense to create this dedicated folder in `Generate Validation Tests`, and to simply output shaders to the source folder in these cross-platform Python tests.

Reply:
Mostly I did this as it otherwise "pollutes" the Python directory and when run locally it shows up as git changes.

Reply:
I see other examples above, though, where we're already modifying files from Materials/TestSuite in place.  For example, the call to mxupdate.py at the top of this block.  Perhaps we could write the output shaders to a similarly safe location in the Materials folder?

Reply:
I've changed the behavour so the default output path is the path of the input document. This should
match behaviour for other utilities. The CI script then does not need this artificial creation of another directory.

Reply:
Thank you, Bernard!